### PR TITLE
[1.20.x] KOGITO-6929 - Upgrade Keycloack image

### PR DESCRIPTION
**Pull request by cristianonicolai**
Depends on https://github.com/kiegroup/kogito-runtimes/pull/2128

### CODE REVIEW

**Review by evacchi — APPROVED** [review]

File: kogito-quarkus-examples/kogito-travel-agency/extended/docker-compose/docker-compose.yml [hunk 1]
```diff
@@ -74,7 +74,7 @@ services:
 
   keycloak:
     container_name: keycloak
-    image: jboss/keycloak
+    image: quay.io/keycloak/keycloak:${KEYCLOAK_VERSION}-legacy
```
Comment (nmirasch) [inline kogito-quarkus-examples/kogito-travel-agency/extended/docker-compose/docker-compose.yml:77]: @cristianonicolai I'm having issues to find this image:quay.io/keycloak/keycloak:14.0.0-legacy.. It worked without the '-legacy' reference.  Is there any issue about not picking that legacy one?

Comment (nmirasch) [reply kogito-quarkus-examples/kogito-travel-agency/extended/docker-compose/docker-compose.yml:77]: @cristianonicolai I see in the related PR there is <version.org.keycloak>17.0.1</version.org.keycloak>, let me give this another try.

Comment (nmirasch) [reply kogito-quarkus-examples/kogito-travel-agency/extended/docker-compose/docker-compose.yml:77]: @cristianonicolai  this works. I guess this PR needs to be linked with https://github.com/kiegroup/kogito-runtimes/pull/2128

Comment (cristianonicolai) [reply kogito-quarkus-examples/kogito-travel-agency/extended/docker-compose/docker-compose.yml:77]: @nmirasch correct

**Review by nmirasch — APPROVED** [review]
Thanks!

**Pull request merged**